lib/homepage_helper.rb

```diff
@@ -1,8 +1,8 @@
 # frozen_string_literal: true
 
 class HomepageHelper
-  def self.resolve(theme_id = nil, current_user = nil)
-    return "custom" if ThemeModifierHelper.new(theme_ids: theme_id).custom_homepage
+  def self.resolve(request = nil, current_user = nil)
+    return "custom" if ThemeModifierHelper.new(request: request).custom_homepage
```

Comment:
This is the gist of the fix. By passing the request, we're ensuring that the modifier helper uses the ids of the theme AND of the components attached to it.